client/blocks/comment-detail/docs/comment-faker.jsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { keyBy, omit } from 'lodash';
+
+/**
+ * `CommentFaker` is a HOC to easily test the Comments Management without the necessity of real data or actions.
+ *
+ * It takes the comments passed via props (with mock data, Redux-connected, etc.) and puts them in its internal state.
+ * Then passes them and various actions (status and like setters) to the actual comment list component.
+ *
+ * Once data and actions will be completely Reduxified, it will be enough to remove the `CommentFaker` call
+ * and use the `mapDispatchToProps` actions instead, leaving this component still useful for testing purposes.
+ */
+export const CommentFaker = WrappedCommentList => class extends Component {
+	state = {
+		comments: {},
+	};
+
+	componentWillMount() {
+		if ( this.props.comments.length ) {
+			this.setState( { comments: keyBy( this.props.comments, 'ID' ) } );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! this.props.comments.length ) {
+			this.setState( { comments: keyBy( nextProps.comments, 'ID' ) } );
+		}
+	}
```

Comment:
fun tip!

I don't always aggressively modularize when I only do a thing twice in code, but in places where I'm duplicating update logic I tend to find value in leaving all of that in one spot, preferring to write duplicate calls over duplicate changes, even if it _adds_ more code overall, with the idea that it lowers our surface-area for update bugs. do as you like since this component is small (now) and it's easy enough to see both places (though it's not obvious that they should both be doing the same thing)

```js
componentWillMount() {
	this.pullCommentsFrom( this.props );
}

componentWillReceiveProps( nextProps ) {
	this.pullCommentsFrom( nextProps );
}

pullCommentsFrom = ( { comments } ) => this.setState( { comments: keyBy( comments, 'ID' ) } );
```

we see at this point too the structure inherent in both former calls and can harness that similarity